packages/squiggle-lang/src/js/index.ts

```diff
@@ -99,25 +104,74 @@ export type squiggleExpression =
 export function run(
   squiggleString: string,
   bindings?: externalBindings,
-  samplingInputs?: samplingParams
+  samplingInputs?: samplingParams,
+  parameters?: parameters
 ): result<squiggleExpression, errorValue> {
-  let b = bindings ? bindings : {};
+  let b = bindings ? bindings : defaultBindings;
+  let p = parameters ? parameters : defaultParameters;
   let si: samplingParams = samplingInputs
     ? samplingInputs
     : defaultSamplingInputs;
 
   let result: result<expressionValue, errorValue> =
-    evaluateUsingExternalBindings(squiggleString, b);
+    evaluateUsingExternalBindings(squiggleString, mergeParameters(b, p));
   return resultMap(result, (x) => createTsExport(x, si));
 }
 
 // Run Partial. A partial is a block of code that doesn't return a value
 export function runPartial(
   squiggleString: string,
-  bindings: externalBindings,
-  _samplingInputs?: samplingParams
+  bindings?: externalBindings,
+  _samplingInputs?: samplingParams,
+  parameters?: parameters
 ): result<externalBindings, errorValue> {
-  return evaluatePartialUsingExternalBindings(squiggleString, bindings);
+  let b = bindings ? bindings : defaultBindings;
+  let p = parameters ? parameters : defaultParameters;
+
+  return evaluatePartialUsingExternalBindings(
+    squiggleString,
+    mergeParameters(b, p)
+  );
+}
+
+function mergeParameters(
+  bindings: externalBindings,
+  parameters: parameters
+): externalBindings {
+  let transformedParemeters = Object.fromEntries(
+    Object.entries(parameters).map(([key, value]) => [
```

Comment:
Maybe you could replace some of this with a nice lodash method?

Reply:
There isn't a clean lodash method for this, I checked. [StackOverflow](https://stackoverflow.com/questions/45358241/lodash-map-keys-and-values-on-an-object) recommends you can use transform, but I tried and to use it I would need to do a whole lot of weird typescript casting (at least on first glance) . Then recommend this method as an option without lodash.